src/petclaw/io/hdf5.py

```diff
@@ -0,0 +1,260 @@
+    for (k,v) in option_defaults.iteritems():
+        options[k] = options.get(k,v)
+    
+    filename = os.path.join(path,'%s%s.hdf' % 
+                                (file_prefix,str(frame).zfill(4)))
+
+    if options['compression'] is not None:
+        err_msg = "Compression (filters) are not available for parallel h5py yet."
+        logging.critical(err_msg)
+        raise Exception(err_msg)
+    
+    if use_h5py:
+        with h5py.File(filename,'w',driver='mpio',comm=MPI.COMM_WORLD) as f:
+
+            # For each patch, write out attributes
+            for state in solution.states:
+                patch = state.patch
+                # Create group for this patch
+                subgroup = f.create_group('patch%s' % patch.patch_index)
+
+                # General patch properties
+                subgroup.attrs['t'] = state.t
+                subgroup.attrs['num_eqn'] = state.num_eqn
+                subgroup.attrs['num_aux'] = state.num_aux
+                for attr in ['num_ghost','patch_index','level']:
+                    if hasattr(patch,attr):
+                        if getattr(patch,attr) is not None:
+                            subgroup.attrs[attr] = getattr(patch,attr)
+
+                # Add the dimension names as a attribute
+                subgroup.attrs['dimensions'] = patch.get_dim_attribute('name')
+                # Dimension properties
+                for dim in patch.dimensions:
+                    for attr in ['num_cells','lower','delta','upper',
+                                 'units']:
+                        if hasattr(dim,attr):
+                            if getattr(dim,attr) is not None:
+                                attr_name = '%s.%s' % (dim.name,attr)
+                                subgroup.attrs[attr_name] = getattr(dim,attr)
+
+                if write_p:
+                    q = state.p
+                else:
+                    q = state.q
+                r = patch._da.getRanges()
+                globalSize = []
+                globalSize.append(q.shape[0])
+                globalSize.extend(patch.num_cells_global)
+                dset = subgroup.create_dataset('q',globalSize,dtype='float',**options)
+                if len(patch.name) == 1:
+                    dset[:,r[0][0]:r[0][1]] = q
+                elif len(patch.name) == 2:
+                    dset[:,r[0][0]:r[0][1],r[1][0]:r[1][1]] = q
+                elif len(patch.name) == 3:
+                    dset[:,r[0][0]:r[0][1],r[1][0]:r[1][1],r[2][0]:r[2][1]] = q
+
+                if write_aux and state.num_aux > 0:
+                    r = patch._da.getRanges()
+                    globalSize = []
+                    globalSize.append(state.num_aux)
+                    globalSize.extend(patch.num_cells_global)
+                    dset = subgroup.create_dataset('aux',globalSize,dtype='float',**options)
+                    if len(patch.name) == 1:
```

Comment:
Done. For example, check https://github.com/aymkhalil/pyclaw/blob/pyclaw_hdf5_parallel_write/src/petclaw/io/hdf5.py#L200.